src/main/java/org/broadinstitute/consent/http/resources/DacResource.java

```diff
@@ -206,6 +213,36 @@ public Response filterUsers(@PathParam("term") String term) {
         return Response.ok().entity(users).build();
     }
 
+    @PUT
+    @Consumes("application/json")
+    @Produces("application/json")
+    @Path("{dacId}/dataset/{datasetId}")
+    @RolesAllowed({CHAIRPERSON})
+    public Response approveDataset(@Auth AuthUser authUser, @PathParam("dacId") Integer dacId, @PathParam("datasetId") Integer datasetId, String json) {
+        try{
+            User user = userService.findUserByEmail(authUser.getEmail());
+            Dataset dataset = datasetService.findDatasetById(datasetId);
+            if(dataset.getDacId() != dacId) { 
+                //Vague message is intentional, don't want to reveal too much info
+                throw new NotFoundException("Dataset not found");
+            }
+            user.checkIfUserHasRole(UserRoles.CHAIRPERSON.getRoleName(), dacId);
+            JsonObject payload = new Gson().fromJson(json, JsonObject.class);
+            if(payload.size() == 0) {
+                throw new BadRequestException("Request body is empty");
+            }
```

Comment:
Super minor optimization - you can check the string length before coercing to a `JsonObject` which would allow you to fail faster in this case.